### PR TITLE
[Feature/Complete] Login through Google (w/ logout)

### PR DESCRIPTION
**Pull request by ytsheng**
- removed feature flag
- added logout
- cleaned up DataStore.jsx

screen shot 2018-05-23 at 9 41 01 pm

### CODE REVIEW

**Comment by ytsheng** [comment]
1. I don't think names should be human readable. That's unscalable.
2. Later PR.

**Comment by ytsheng** [comment]
1. What if they don't have unix names? A hash is way easier.
2. Creating a new person upon login as a user is a new feature. Leaving in the flag is messy, this does not break functionality.

**Comment by ytsheng** [comment]
I agree google uid probably shouldn't be somebody's id, but i disagree on the id needing to be readable.

**Comment by bzreinhardt** [comment]
http://friendforce.app/person/sasha_sheng123 is way better than http://friendforce.app/person/2ghf1-298hr20338f3h3ge17

Beautiful URLs are important

**Comment by ytsheng** [comment]
I see, forgot about the fact that it is being used as a route.

File: my-app/src/App.js [hunk 1]
```diff
@@ -57,28 +57,37 @@ class App extends Component {
...
-      this.login = this.login.bind(this);
-      this.logout = this.logout.bind(this);
-    }
+    this.login = this.login.bind(this);
+    this.logout = this.logout.bind(this);
   }
 
   login() {
     auth.signInWithPopup(provider).then((result) => {
       const user = result.user;
```
Comment (bzreinhardt) [inline my-app/src/App.js]: Blah blah

**Pull request merged**